scripts/get-nightly-build-pass-rate.ts

```diff
@@ -0,0 +1,108 @@
+import puppeteer from 'puppeteer';
+import dotenv from 'dotenv';
+dotenv.config();
+
+const loginAndAnalyzeBuilds = async (
+  latestBuildNumber: number,
+  numberOfDays: number,
```

Comment:
I think we're more likely to have the start date of a PI on hand rather than the number of days back we want to go. Do you think you could change this to accept a PI start date?

Reply:
I think accepting a date as input might introduce some challenges. Are you suggesting that the user manually identify the build number for the start of the PI based on that date? I feel that could make things more difficult, especially since we'll likely be running this script after a PI concludes.

It seems easier to locate the build number for the end of the PI—jenkins and our slack channel make it straightforward to find the most recent build. Backtracking to determine the exact build corresponding to a PI's start date would likely require more effort on the user's part. That said, if there’s an automated way to map dates to build numbers, I’d be happy to explore that!

Reply:
I'm also asking you to work with Dates and Time in _Javascript_, so it's ok if that's just a categorical 'no' 😆

Reply:
@jack-ryan-nava-pbc, Ohhh I misunderstood! I think I see the appeal in using a specific date to calculate the starting build number—it definitely aligns more naturally with how we think about time. However, since our PIs are typically a fixed duration (I should double check this assumption), sticking to a set number of days might keep things simpler and more predictable for our use case?

Reply:
Well, a fixed number of days wouldn't work if the PI isn't complete yet?

Reply:
> Well, a fixed number of days wouldn't work if the PI isn't complete yet right?

Ah, I see your point now! I think I might not have been super clear earlier. When I mentioned a set number of days, I was referring to the fact that PIs are typically a fixed duration (e.g., 54 days for PI4). Since this script is mostly being used in retrospect, I thought it might be easier to find the final build number and work backward a set number of builds, rather than trying to pinpoint the exact date for the start of the PI.

If you feel strongly about using dates, let’s move this discussion over to [pr-75](https://github.cms.gov/CMS-WDS/experience-toolkit/pull/75) since I’m planning to close this one out.